### DISCUSSION TITLE
Which are the diff between a released firmware and the compiled one?

### DISCUSSION THREAD

**Discussion by MauroDiamantino** [micropython/micropython#12764] [ESP32]
Hi, I'm seeing differences in size and behavior between a released firmware (published [here](https://micropython.org/download/ESP32_GENERIC/)) and the firmware I get if I compile the git project with idf.py v5.0.2. Taking into account, the same micropython version (v1.21.0) and the same variant (generic). Are the differences in the esp-idf version?
I've enabled flash encryption in the ESP32 (I'm using a ESP-WROOM-32 board with a ESP32-D0WD-V3 chip) and I'm pre-encrypting the micropython firmware and then I write in the board. Both firmware work well but the compiled version isn't able to access the data partition. I get `OSError: [Errno 19] ENODEV` when I try to open a file.

**Comment by jimmo** [comment]
How much of a size difference? Are you building at the release tag?

Probably the most useful way to compare the two builds is with:

```
>>> import platform
>>> platform.python_compiler()
'GCC 11.2.0'
>>> platform.platform()
'MicroPython-1.20.0-xtensa-IDFv5.0.2-with-newlib4.1.0'
>>> platform.libc_ver()
('newlib', '4.1.0')
```

(That's just the output for a random board on my desk, but at least will let you compare how the release and the local firmware are made)

**Comment by bikeNomad** [comment]
How are you building and encrypting the data partition?
